src/include/indexlr-workers.h

```diff
@@ -338,16 +332,22 @@ MinimizeWorker::work()
 
 			auto minimizers = getMinimizers(hashes, w);
 
-			if (stlfr) {
-				std::regex right("^.*#+");
-				std::regex left("/.*");
-
-				read.barcode = std::regex_replace(read.id, right, "");
-				read.barcode = std::regex_replace(read.barcode, left, "");
+			if (read.id.find("BX:Z:") == std::string::npos) {
+				size_t rightPos = read.id.find('#');
+				if (rightPos == std::string::npos) {
+					ss << read.barcode;
```

Comment:
`read.barcode` should be empty here. What's the purpose of this line?

Reply:
Before I implemented this, `ss << read.barcode` will execute regardless of if the barcode is empty so I thought I should keep that

Reply:
The line of code is necessary in order to output NA for reads that have no barcode in the read

Reply:
This logic of detecting the stLFR barcode should be moved up to where the `barcode` member is initialized at
https://github.com/bcgsc/physlr/blob/f9665f5392fe70068c95318cb06efe32f5f240ed/src/include/indexlr-workers.h#L286